pkg/api/util/ipc_endpoint.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package util
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/system"
+)
+
+// IPCEndpoint is an endpoint that IPC requests will be sent to
+type IPCEndpoint struct {
+	client    *http.Client
+	target    url.URL
+	closeConn bool
+}
+
+// GetOption is an option that can be passed to DoGet
+type GetOption func(url.URL) url.URL
+
+// WithValues is an option to add url.Values to the GET request
+func WithValues(values url.Values) GetOption {
+	return func(u url.URL) url.URL {
+		u.RawQuery = values.Encode()
+		return u
+	}
+}
+
+// DoGet sends GET method to the endpoint
+func (end *IPCEndpoint) DoGet(options ...GetOption) ([]byte, error) {
```

Comment:
I think we can move the option `WithValues` to the constructor and avoid having different option constructor for different functions. 

```go
func WithValues(values url.Values) GetOption {
	return func(end *IPCEndpoint) url.URL {
		end.target.RawQuery = values.Encode()
		return u
	}
}
```

That way we do not have different options depending on the function that gets call and all configuration of the IPCEndpoint happens at the `NewIPCEndpoint` function.

WDYT?

Reply:
I had it like that before [this commit](https://github.com/DataDog/datadog-agent/pull/22886/commits/bc848fa3e570df1236ffa6cb650545dbd9e1438c), but decided to change it to this. My reasoning: soon there will be a `DoPost` method (needed for [pkg/diagnose/runner.go](https://github.com/DataDog/datadog-agent/blob/199d74cb775d8bd3d8cc015e1a522ee760c03e24/pkg/diagnose/runner.go#L322)) and post parameters should be passed to that method rather than placed in the constructor. The `url.values` are the query parameters for `GET` methods, making them analogous to the body of `POST`. The endpoint constructor params should concern the endpoint itself: its host, port, connection type, etc. Params given to `DoGet` and `DoPost` should concern what is sent in the request itself.